xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -184,376 +181,373 @@ bool CEdl::ReadEdl(const std::string& strMovie, const float fFramesPerSecond)
+                   CURL::GetRedacted(edlFilename));
         continue;
       }
       break;
     case 3:
       edit.action = Action::COMM_BREAK;
       if (!AddEdit(edit))
       {
-        CLog::Log(LOGWARNING, "{} - Error adding commercial break from line {} in EDL file: {}",
-                  __FUNCTION__, iLine, CURL::GetRedacted(edlFilename));
+        CLog::LogF(LOGWARNING, "Error adding commercial break from line {} in EDL file: {}", line,
+                   CURL::GetRedacted(edlFilename));
         continue;
       }
       break;
     default:
-      CLog::Log(LOGWARNING, "{} - Invalid action on line {} in EDL file: {}", __FUNCTION__, iLine,
-                CURL::GetRedacted(edlFilename));
+      CLog::LogF(LOGWARNING, "Invalid action on line {} in EDL file: {}", line,
+                 CURL::GetRedacted(edlFilename));
       continue;
     }
   }
 
-  if (bError) // Log last line warning, if there was one, since while loop will have terminated.
-    CLog::Log(LOGWARNING, "{} - Error on line {} in EDL file: {}", __FUNCTION__, iLine,
-              CURL::GetRedacted(edlFilename));
+  // Log last line warning, if there was one, since while loop will have terminated.
+  if (error)
+    CLog::Log(LOGWARNING, "Error on line {} in EDL file: {}", line, CURL::GetRedacted(edlFilename));
 
   edlFile.Close();
 
   if (HasEdits() || HasSceneMarker())
   {
-    CLog::Log(LOGDEBUG, "{0} - Read {1} edits and {2} scene markers in EDL file: {3}", __FUNCTION__,
-              m_vecEdits.size(), m_vecSceneMarkers.size(), CURL::GetRedacted(edlFilename));
+    CLog::LogF(LOGDEBUG, "Read {} edits and {} scene markers in EDL file: {}", m_edits.size(),
+               m_sceneMarkers.size(), CURL::GetRedacted(edlFilename));
     return true;
   }
   else
   {
-    CLog::Log(LOGDEBUG, "{} - No edits or scene markers found in EDL file: {}", __FUNCTION__,
-              CURL::GetRedacted(edlFilename));
+    CLog::LogF(LOGDEBUG, "No edits or scene markers found in EDL file: {}",
+               CURL::GetRedacted(edlFilename));
     return false;
   }
 }
 
-bool CEdl::ReadComskip(const std::string& strMovie, const float fFramesPerSecond)
+bool CEdl::ReadComskip(const std::string& path, float fps)
 {
   Clear();
 
-  std::string comskipFilename(URIUtils::ReplaceExtension(strMovie, ".txt"));
+  std::string comskipFilename{URIUtils::ReplaceExtension(path, ".txt")};
   if (!CFile::Exists(comskipFilename))
     return false;
 
   CFile comskipFile;
   if (!comskipFile.Open(comskipFilename))
   {
-    CLog::Log(LOGERROR, "{} - Could not open Comskip file: {}", __FUNCTION__,
-              CURL::GetRedacted(comskipFilename));
+    CLog::LogF(LOGERROR, "Could not open Comskip file: {}", CURL::GetRedacted(comskipFilename));
     return false;
   }
 
-  char szBuffer[1024];
-  if (comskipFile.ReadString(szBuffer, 1023)
-  &&  strncmp(szBuffer, COMSKIP_HEADER, strlen(COMSKIP_HEADER)) != 0) // Line 1.
+  std::array<char, 1024> buffer{};
```

Comment:
Do you know where this value comes from?

Reply:
big enough buffer size, to be able to use IFile::Read we need to provide a size unfortunately